test/modules/layers/geojson-layer.spec.js

```diff
@@ -126,7 +128,107 @@ test('GeoJsonLayer#tests', t => {
     },
     props: {
       // TODO: Set a right geojson example as the provided from 'deck.gl-data' contains 'GeometryCollection' types that are not compatible with geojsonToBinary
-      data: geojsonToBinary(binaryData)
+      data: binaryData
+    }
+  });
+
+  const binaryDataWithAttributes = {
+    ...binaryData,
+    points: {
+      ...binaryData.points,
+      attributes: {
+        getRadius: {
+          size: 1,
+          value: binaryData.points.featureIds.value.map(_ => Math.round(Math.random() * 10))
+        }
+      }
+    },
+    lines: {
+      ...binaryData.lines,
+      attributes: {
+        getWidth: {
+          size: 1,
+          value: binaryData.lines.featureIds.value.map(_ => Math.random())
+        }
+      }
+    },
+    polygons: {
+      ...binaryData.polygons,
+      attributes: {
+        getColor: {
+          size: 3,
+          value: new Uint16Array(binaryData.polygons.featureIds.value.length * 3).fill(255)
+        }
+      }
+    }
+  };
+
+  testCases.push({
+    title: 'GeoJsonLayer#binaryAttributes',
+    onBeforeUpdate: ({testCase}) => t.comment(testCase.title),
+    onAfterUpdate: ({layer, subLayers, subLayer}) => {
+      // Polygons-fill
+      t.ok(
+        subLayers[0].props.data.attributes.getColor,
+        'polygon-fill subLayer should receive passed binary attribute'
+      );
+      // Polygons-stroke
+      t.ok(
+        subLayers[1].props.data.attributes.getColor,
+        'polygon-stroke subLayer should receive passed binary attribute'
+      );
+      // Lines
+      t.ok(
+        subLayers[2].props.data.attributes.getWidth,
+        'linestrings subLayer should receive passed binary attribute'
+      );
+      // Points
+      t.ok(
+        subLayers[3].props.data.attributes.getRadius,
+        'points subLayer should receive passed binary attribute'
+      );
+    },
+    props: {
+      // TODO: Set a right geojson example as the provided from 'deck.gl-data' contains 'GeometryCollection' types that are not compatible with geojsonToBinary
+      data: binaryDataWithAttributes
+    }
+  });
+
+  const binaryDataWithGetFilterValue = {
+    ...binaryDataWithAttributes,
```

Comment:
Why extend `binaryDataWithAttributes` here rather than just `binaryData`? Makes for a more isolated test

Reply:
Ok, better 👌